README.md
## get_dialog_service

This package provides < Get Dialog Service >.

You can show an alert dialog just call the method in this service widget.

## Sample Img

![](https://raw.githubusercontent.com/strong1133/get_dialog_service/master/img/sample1.png)

![](https://raw.githubusercontent.com/strong1133/get_dialog_service/master/img/sample2.png)

![](https://raw.githubusercontent.com/strong1133/get_dialog_service/master/img/sample3.png)


## Doc

- [GitHub Repository](https://github.com/strong1133/get_dialog_service)

## Usage
In your project, must use GetMaterialApp widget 



CHANGELOG.md
## 1.0.0

* Initial release.
* Success, Fail, Warn, Confirm 


## 1.0.1

* Add Loading Dialog

test/get_dialog_service_test.dart
import 'package:flutter_test/flutter_test.dart';

import 'package:get_dialog_service/get_dialog_service.dart';

void main() {
  test('adds one to input values', () {
    final calculator = Calculator();
    expect(calculator.addOne(2), 3);
    expect(calculator.addOne(-7), -6);
    expect(calculator.addOne(0), 1);
  });
}


example/lib/home.dart
import 'package:flutter/material.dart';
import 'package:get/get.dart';
import 'package:get_dialog_service/dialog_service.dart';

class Home extends StatelessWidget {
  const Home({Key? key}) : super(key: key);

  @override
  Widget build(BuildContext context) {
    return Scaffold(
      appBar: _buildAppBar(),
      body: _buildBody(),
    );
  }

  AppBar _buildAppBar() {
    return AppBar(
      title: const Text('Get Dialog Service Example'),
    );
  }

  Widget _buildBody() {
    return SizedBox(
      width: double.maxFinite,
      height: double.maxFinite,
      child: Column(
        children: [
          spacer(height: 30),
          Padding(
            padding: const EdgeInsets.all(20.0),
            child: Column(
              children: [
                const Text(
                  'Test Buttons',
                  style: TextStyle(fontSize: 20, fontWeight: FontWeight.bold),
                ),
                spacer(height: 20),
                Padding(
                  padding: const EdgeInsets.all(15.0),
                  child: ElevatedButton(
                      onPressed: () {
                        // ignore: avoid_print
                        print('Call Success Dialog');
                        DialogService().getSuccess(
                            width: 200,
                            msg: 'Success',
                            lang: 'ko',
                            callback: () {
                              // ignore: avoid_print
                              print('Call Back Success Dialog!!');
                            });
                      },
                      child: const Text('Call Success Dialog')),
                ),
                Padding(
                  padding: const EdgeInsets.all(15.0),
                  child: ElevatedButton(
                      style: ElevatedButton.styleFrom(
                        primary: Colors.red,
                      ),
                      onPressed: () {
                        // ignore: avoid_print
                        print('Call Fail Dialog');
                        DialogService().getFail(
                            width: 200,
                            msg: 'Fail on Error',
                            lang: 'ko',
                            callback: () {
                              // ignore: avoid_print
                              print('Call Back Fail Dialog!!');
                            });
                      },
                      child: const Text('Call Fail Dialog')),
                ),
                Padding(
                  padding: const EdgeInsets.all(15.0),
                  child: ElevatedButton(
                      style: ElevatedButton.styleFrom(
                        primary: Colors.yellow[600],
                      ),
                      onPressed: () {
                        // ignore: avoid_print
                        print('Call Warning Dialog');
                        DialogService().getWarn(
                            width: 200,
                            msg: 'Warning',
                            lang: 'ko',
                            callback: () {
                              // ignore: avoid_print
                              print('Call Back Warning Dialog!!');
                            });
                      },
                      child: const Text('Call Warning Dialog')),
                ),
                Padding(
                  padding: const EdgeInsets.all(15.0),
                  child: ElevatedButton(
                      style: ElevatedButton.styleFrom(
                        primary: Colors.deepPurple,
                      ),
                      onPressed: () {
                        // ignore: avoid_print
                        print('Call Confirm Dialog');
                        DialogService().getConfirm(
                            width: 200,
                            msg: 'Confirm',
                            lang: 'ko',
                            callback: () {
                              // ignore: avoid_print
                              print('Call Back Confirm Dialog!! -> OK!! ');
                            },
                            noCallback: () {
                              // ignore: avoid_print
                              print('Call Back Confirm Dialog!! -> No!! ');
                            });
                      },
                      child: const Text('Call Confirm Dialog')),
                ),
                Padding(
                  padding: const EdgeInsets.all(15.0),
                  child: ElevatedButton(
                      style: ElevatedButton.styleFrom(
                        primary: Colors.brown[400],
                      ),
                      onPressed: () {
                        // ignore: avoid_print
                        print('Call Loading Dialog');
                        DialogService().getLoaidng();
                        Future.delayed(const Duration(seconds: 3), () {
                          // deleayed code here
                          Get.back();
                        });
                      },
                      child: const Text('Call Loading Dialog')),
                ),
                spacer(height: 10),
                Padding(
                  padding: const EdgeInsets.all(15.0),
                  child: ElevatedButton(
                      style: ElevatedButton.styleFrom(
                        primary: Colors.amber[800],
                      ),
                      onPressed: () {
                        // ignore: avoid_print
                        print('Call Confirm Dialog');
                        DialogService().getConfirm(
                            msg: "Processing?",
                            callback: () {
                              DialogService().getLoaidng();
                              Future.delayed(const Duration(seconds: 3), () {
                                Get.back();
                              }).then((value) => DialogService().getSuccess(msg: "Success Process"));
                            },
                            noCallback: () {
                              DialogService().getFail(msg: "Exit Process");
                            });

                        // DialogService().getLoaidng();
                        Future.delayed(const Duration(seconds: 3), () {
                          // deleayed code here
                          Get.back();
                        });
                      },
                      child: const Text('Confirm -> Loading -> Success')),
                ),
              ],
            ),
          ),
        ],
      ),
    );
  }

  SizedBox spacer({required double height}) {
    return SizedBox(
      height: height,
    );
  }
}


lib/dialog_service.dart
import 'package:get/get.dart';
import 'package:get_dialog_service/call_back_dialog.dart';

class DialogService extends GetxService {
  CallBackDialog callBackDialog = CallBackDialog();

  // Call Success Dialog
  void getSuccess({double? width, String? msg, String lang = 'ko', Function? callback}) {
    callBackDialog.getSuccessAltCb(
        width: width,
        msg: msg,
        lang: lang,
        callBack: () {
          if (callback != null) {
            callback();
          } else {
            // ignore: avoid_print
            print("NO CALLBACK");
          }
        });
  }

  // Call Fail Dialog
  void getFail({double? width, String? msg, String lang = 'ko', Function? callback}) {
    callBackDialog.getFailAltCb(
        width: width,
        msg: msg,
        lang: lang,
        callBack: () {
          if (callback != null) {
            callback();
          } else {
            // ignore: avoid_print
            print("NO CALLBACK");
          }
        });
  }

  // Call Warinig Dialog
  void getWarn({double? width, String? msg, String lang = 'ko', Function? callback}) {
    callBackDialog.getWarnAltCb(
        width: width,
        msg: msg,
        lang: lang,
        callBack: () {
          if (callback != null) {
            callback();
          } else {
            // ignore: avoid_print
            print("NO CALLBACK");
          }
        });
  }

  // Call Confirm Dialog
  void getConfirm({double? width, String? msg, String lang = 'ko', Function? callback, Function? noCallback}) {
    callBackDialog.getConfirmAltCb(
        width: width,
        msg: msg,
        lang: lang,
        callBack: () {
          if (callback != null) {
            callback();
          } else {
            // ignore: avoid_print
            print("NO CALLBACK");
          }
        },
        noCallBack: () {
          if (noCallback != null) {
            noCallback();
          } else {
            // ignore: avoid_print
            print("NO CALLBACK");
          }
        });
  }

  // Call Loading Dialog
  void getLoaidng() {
    callBackDialog.getLoadingAltCb();
  }
}


lib/call_back_dialog.dart
import 'package:flutter/material.dart';
import 'package:rflutter_alert/rflutter_alert.dart';
import 'package:get/get.dart';

class CallBackDialog {
  //
  // Get Success Alert Diaglog with Call Back Func
  void getSuccessAltCb({double? width, String? title, String? msg, String? lang, required Function callBack}) {
    // ignore: no_leading_underscores_for_local_identifiers
    String _tempMsg = '';
    // ignore: no_leading_underscores_for_local_identifiers
    String _tempTitle = '';

    _tempMsg = msg ?? 'Success';
    _tempTitle = title ?? (lang == 'ko' ? '성공' : 'Success');

    Get.dialog(AlertDialog(
      title: Text(
        _tempTitle,
        style: const TextStyle(fontSize: 20, fontWeight: FontWeight.w700),
        textAlign: TextAlign.center,
      ),
      content: SingleChildScrollView(
        child: SizedBox(
          width: width ?? 200,
          child: ListBody(
            //List Body를 기준으로 Text 설정a
            children: <Widget>[
              Center(
                child: Image.asset(
                  '$kImagePath/icon_success.png',
                  package: 'rflutter_alert',
                  width: 50,
                  height: 50,
                ),
              ),
              const SizedBox(height: 40),
              Text(
                _tempMsg,
                style: const TextStyle(fontSize: 20, fontWeight: FontWeight.w400),
                textAlign: TextAlign.center,
              ),
            ],
          ),
        ),
      ),
      actions: [
        Column(
          children: [
            Align(
              alignment: Alignment.center,
              child: ElevatedButton(
                style: ElevatedButton.styleFrom(
                  elevation: 0,
                  minimumSize: const Size(100, 40),
                ),
                child: Text(
                  lang == 'ko' ? '확인' : 'Ok',
                  textAlign: TextAlign.center,
                ),
                onPressed: () {
                  Get.back();
                  callBack();
                },
              ),
            ),
            const SizedBox(height: 20),
          ],
        ),
      ],
    ));
  }

  //
  // Get Fail Alert Diaglog with Call Back Func
  void getFailAltCb({double? width, String? title, String? msg, String? lang, required Function callBack}) {
    // ignore: no_leading_underscores_for_local_identifiers
    String _tempMsg = '';
    // ignore: no_leading_underscores_for_local_identifiers
    String _tempTitle = '';

    _tempMsg = msg ?? 'Fail';
    _tempTitle = title ?? (lang == 'ko' ? '실패' : 'Fail');

    Get.dialog(AlertDialog(
      title: Text(
        _tempTitle,
        style: const TextStyle(fontSize: 20, fontWeight: FontWeight.w700),
        textAlign: TextAlign.center,
      ),
      content: SingleChildScrollView(
        child: SizedBox(
          width: width ?? 200,
          child: ListBody(
            //List Body를 기준으로 Text 설정a
            children: <Widget>[
              Center(
                child: Image.asset(
                  '$kImagePath/icon_error.png',
                  package: 'rflutter_alert',
                  width: 50,
                  height: 50,
                ),
              ),
              const SizedBox(height: 40),
              Text(
                _tempMsg,
                style: const TextStyle(fontSize: 20, fontWeight: FontWeight.w400),
                textAlign: TextAlign.center,
              ),
            ],
          ),
        ),
      ),
      actions: [
        Column(
          children: [
            Align(
              alignment: Alignment.center,
              child: ElevatedButton(
                style: ElevatedButton.styleFrom(
                  elevation: 0,
                  minimumSize: const Size(100, 40),
                ),
                child: Text(
                  lang == 'ko' ? '확인' : 'Ok',
                  textAlign: TextAlign.center,
                ),
                onPressed: () {
                  Get.back();
                  callBack();
                },
              ),
            ),
            const SizedBox(height: 20),
          ],
        ),
      ],
    ));
  }

  //
  // Get Warning Alert Diaglog with Call Back Func
  void getWarnAltCb({double? width, String? title, String? msg, String? lang, required Function callBack}) {
    // ignore: no_leading_underscores_for_local_identifiers
    String _tempMsg = '';
    // ignore: no_leading_underscores_for_local_identifiers
    String _tempTitle = '';

    _tempMsg = msg ?? 'Warning';
    _tempTitle = title ?? (lang == 'ko' ? '경고' : 'Warning');

    Get.dialog(AlertDialog(
      title: Text(
        _tempTitle,
        style: const TextStyle(fontSize: 20, fontWeight: FontWeight.w700),
        textAlign: TextAlign.center,
      ),
      content: SingleChildScrollView(
        child: SizedBox(
          width: width ?? 200,
          child: ListBody(
            //List Body를 기준으로 Text 설정a
            children: <Widget>[
              Center(
                child: Image.asset(
                  '$kImagePath/icon_warning.png',
                  package: 'rflutter_alert',
                  width: 50,
                  height: 50,
                ),
              ),
              const SizedBox(height: 40),
              Text(
                _tempMsg,
                style: const TextStyle(fontSize: 20, fontWeight: FontWeight.w400),
                textAlign: TextAlign.center,
              ),
            ],
          ),
        ),
      ),
      actions: [
        Column(
          children: [
            Align(
              alignment: Alignment.center,
              child: ElevatedButton(
                style: ElevatedButton.styleFrom(
                  elevation: 0,
                  minimumSize: const Size(100, 40),
                ),
                child: Text(
                  lang == 'ko' ? '확인' : 'Ok',
                  textAlign: TextAlign.center,
                ),
                onPressed: () {
                  Get.back();
                  callBack();
                },
              ),
            ),
            const SizedBox(height: 20),
          ],
        ),
      ],
    ));
  }

  //
  // Get Warning Alert Diaglog with Call Back Func
  void getConfirmAltCb({double? width, String? title, String? msg, String? lang, required Function callBack, required Function noCallBack}) {
    // ignore: no_leading_underscores_for_local_identifiers
    String _tempMsg = '';
    // ignore: no_leading_underscores_for_local_identifiers
    String _tempTitle = '';

    _tempMsg = msg ?? 'Confirm';
    _tempTitle = title ?? (lang == 'ko' ? '확인' : 'Confirm');

    Get.dialog(AlertDialog(
      title: Text(
        _tempTitle,
        style: const TextStyle(fontSize: 20, fontWeight: FontWeight.w700),
        textAlign: TextAlign.center,
      ),
      content: SingleChildScrollView(
        child: SizedBox(
          width: width ?? 200,
          child: ListBody(
            //List Body를 기준으로 Text 설정a
            children: <Widget>[
              Center(
                child: Image.asset(
                  '$kImagePath/icon_info.png',
                  package: 'rflutter_alert',
                  width: 50,
                  height: 50,
                ),
              ),
              const SizedBox(height: 40),
              Text(
                _tempMsg,
                style: const TextStyle(fontSize: 20, fontWeight: FontWeight.w400),
                textAlign: TextAlign.center,
              ),
            ],
          ),
        ),
      ),
      actions: [
        Column(
          children: [
            Row(mainAxisAlignment: MainAxisAlignment.center, children: [
              ElevatedButton(
                style: ElevatedButton.styleFrom(
                  elevation: 0,
                  minimumSize: const Size(100, 40),
                ),
                child: Text(
                  lang == 'ko' ? '확인' : 'Ok',
                  textAlign: TextAlign.center,
                ),
                onPressed: () {
                  Get.back();
                  callBack();
                },
              ),
              const SizedBox(width: 30),
              ElevatedButton(
                style: ElevatedButton.styleFrom(
                  primary: Colors.red,
                  elevation: 0,
                  minimumSize: const Size(100, 40),
                ),
                child: Text(
                  lang == 'ko' ? '취소' : 'Cancle',
                  textAlign: TextAlign.center,
                ),
                onPressed: () {
                  Get.back();
                  noCallBack();
                },
              ),
            ]),
            const SizedBox(height: 20),
          ],
        ),
      ],
    ));
  }

  //
  // Get Loading Alert Diaglog
  void getLoadingAltCb() {
    Get.dialog(AlertDialog(
      shape: RoundedRectangleBorder(borderRadius: BorderRadius.circular(8.0)),
      content: const SizedBox(
        height: 200,
        child: Center(
            child: SizedBox(
          height: 50.0,
          width: 50.0,
          child: CircularProgressIndicator(valueColor: AlwaysStoppedAnimation(Colors.blue), strokeWidth: 5.0),
        )),
      ),
    ));
  }
}
